index.bs

```diff
@@ -1898,58 +1775,54 @@ unless the user agent also implements the URL [[URL]] specification.
 In other words, <code>URL.prototype</code> must evaluate to true if the user agent implements the URL [[URL]] specification,
 and must NOT evaluate to true otherwise.
 
-<h4 id="createRevokeMethodsParams">
-Methods and Parameters</h4>
+<div algorithm="createObjectURL">
+The <dfn method for=URL id="dfn-createObjectURL">createObjectURL(|blob|)</dfn> static method must run these steps:
 
-<dl>
-  <dt>The <dfn method for=URL id="dfn-createObjectURL">createObjectURL(blob)</dfn> static method
-  <dd>Returns a unique <a>Blob URL</a>.
-    This method must act as follows:
-
-    1. Let |url| be the result of the <a>Serialization of a Blob URL</a> algorithm.
-    2. <a>Add an entry to the Blob URL Store</a> for |url| and |blob|.
-    3. Return |url|.
-
-  <dt>The <dfn method for=URL id="dfn-revokeObjectURL">revokeObjectURL(url)</dfn> static method
-  <dd>Revokes the <a>Blob URL</a> provided in the string {{url!!argument}}
-    by removing the corresponding entry from the <a>Blob URL Store</a>.
-    This method must act as follows:
-
-    1. If the value provided for the {{url!!argument}} argument is not a <a>Blob URL</a>,
-      OR if the value provided for the {{url!!argument}} argument does not have an entry in the <a>Blob URL Store</a>,
-      this method call does nothing.
-      User agents may display a message on the error console.
-    2. Otherwise, user agents must <a>remove the entry</a> from the <a>Blob URL Store</a> for {{url!!argument}}.
-
-      Note: Subsequent attemps to dereference {{url!!argument}} result in a <a>network error</a>,
-      since the entry has been removed from the <a>Blob URL Store</a>.
-
-    The <dfn argument for="URL/revokeObjectURL(url)" id="dfn-urlarg">url</dfn> argument to the {{revokeObjectURL()}} method
-    is a <a>Blob URL</a> string.
-
-    <div class="example">
-      In the example below,
-      <code>window1</code> and <code>window2</code> are separate,
-      but in the <a>same origin</a>;
-      <code>window2</code> could be an <{iframe}> inside <code>window1</code>.
-
-      <pre class="lang-javascript">
-        myurl = window1.URL.createObjectURL(myblob);
-        window2.URL.revokeObjectURL(myurl);
-      </pre>
-
-      Since <code>window1</code> and <code>window2</code> are in the <a>same origin</a>
-      and share the same <a>Blob URL Store</a>,
-      the <code>URL.{{revokeObjectURL()}}</code> call
-      ensures that subsequent dereferencing of <code>myurl</code>
-      results in a the user agent acting as if a <a>network error</a> has occurred.
-    </div>
-</dl>
+1. Let |url| be the result of [=adding an entry to the Blob URL Store=] for |blob|.
+4. Return |url|.
+
+</div>
+
+<div algorithm="revokeObjectURL">
+The <dfn method for=URL id="dfn-revokeObjectURL">revokeObjectURL(|url|)</dfn> static method must run these steps:
+
+1. [=Remove an entry from the Blob URL Store=] for |url|.
```

Comment:
I'm assuming these steps are still being worked on? Looks like this is incomplete. (Redundant first/last steps, FIXME, etc)

Reply:
Actually after testing I decided to do keep the same-origin check, as at least chrome and firefox already seem to do that (well, Firefox seems to have a same-origin check, Chrome only lets you revoke URLs from the context that created them entirely).